File: src/main/java/com/teamwizardry/wizardry/api/util/RandUtil.java
package com.teamwizardry.wizardry.api.util;

import java.util.Random;

/**
 * Created by Demoniaque.
 */
public class RandUtil {

	public static Random random = new Random();

	public static double nextDouble(double min, double max) {
		return (random.nextDouble() * (max - min)) + min;
	}

	public static double nextDouble(double bound) {
		return (random.nextDouble() * (bound));
	}

	public static long nextDouble() {
		return (long) random.nextDouble();
	}

	public static long nextLong(long min, long max) {
		return (long) ((random.nextDouble() * (max - min)) + min);
	}

	public static long nextLong(long bound) {
		return (long) (random.nextDouble() * (bound));
	}

	public static double nextLong() {
		return random.nextDouble();
	}

	public static float nextFloat(float min, float max) {
		return (random.nextFloat() * (max - min)) + min;
	}

	public static float nextFloat(float bound) {
		return (random.nextFloat() * (bound));
	}

	public static float nextFloat() {
		return random.nextFloat();
	}

	public static int nextInt(int min, int max) {
		return (int) ((random.nextDouble() * (max - min)) + min);
	}

	public static int nextInt(int bound) {
		return (int) (random.nextDouble() * bound);
	}

	public static int nextInt() {
		return (int) random.nextDouble();
	}

	public static boolean nextBoolean() {
		return random.nextBoolean();
	}
}

File: src/main/kotlin/com/teamwizardry/wizardry/common/utils/RandUtil.kt
package com.teamwizardry.wizardry.common.utils

import net.minecraft.util.math.Vec3d
import java.util.*

object RandUtil {
    private var random = Random()
    fun nextVec3d(minX: Double, maxX: Double, minY: Double, maxY: Double, minZ: Double, maxZ: Double): Vec3d {
        return Vec3d(nextDouble(minX, maxX), nextDouble(minY, maxY), nextDouble(minZ, maxZ))
    }

    fun nextDouble(min: Double, max: Double): Double {
        return random.nextDouble() * (max - min) + min
    }

    fun nextDouble(bound: Double): Double {
        return random.nextDouble() * bound
    }

    fun nextDouble(): Long {
        return random.nextDouble().toLong()
    }

    fun nextLong(min: Long, max: Long): Long {
        return (random.nextDouble() * (max - min) + min).toLong()
    }

    fun nextLong(bound: Long): Long {
        return (random.nextDouble() * bound).toLong()
    }

    fun nextLong(): Double {
        return random.nextDouble()
    }

    fun nextFloat(min: Float, max: Float): Float {
        return random.nextFloat() * (max - min) + min
    }

    fun nextFloat(bound: Float): Float {
        return random.nextFloat() * bound
    }

    fun nextFloat(): Float {
        return random.nextFloat()
    }

    fun nextInt(min: Int, max: Int): Int {
        return (random.nextDouble() * (max - min) + min).toInt()
    }

    fun nextInt(bound: Int): Int {
        return (random.nextDouble() * bound).toInt()
    }

    fun nextInt(): Int {
        return random.nextDouble().toInt()
    }

    fun nextBoolean(): Boolean {
        return random.nextBoolean()
    }
}